content/chapitres/project-lifecycle.adoc

```diff
@@ -24,122 +24,7 @@ Comment fabriquer et tester de manière reproductible ?
 * Dimension humaine: habitudes, connaissances, documentation (ou pas)
 
-[{invert}]
-== Exemple avec Java ☕️
-
-JAVA bien ?
-
-* Essayons un projet "Hello World" en Java dans GitPod, dans un nouveau dossier vide :
-
-image::https://gitpod.io/button/open-in-gitpod.svg[link="https://gitpod.io/workspaces",window="_blank"]
-
-== Exemple Java : Compiler 🤲🏽
-
-* Créez un fichier source java `main.java` avec le contenu suivant :
-+
-[source,java]
-----
-class HelloWorld {
-    public static void main(String[] args) {
-        System.out.println("Hello ENSG!");
-    }
-}
-----
-
-* La commande `javac` permet de "compiler" du java en bytecode java.
-** Compilez le programme dans un dossier `target/` :
-+
-[source,bash]
-----
-javac ./main.java -d ./target
-ls -ltra ./target # Quel est le résultat ?
-----
-
-== Exemple Java : Exécuter (Artisanalement) 🙈
-
-* Un fichier `HelloWorld.class` a été généré
-** C'est du "bytecode binaire" Java exécutable avec la commande `java`
-
-* Essayez d'exécuter le programme :
-** La commande `java` a besoin de connaître le dossier contenant les classes
-+
-[source,bash]
-----
-java -classpath ./target/ HelloWorld
-----
-
-== Exemple Java : Livrable JAR 🏺
-
-* 🤔 Il faut connaître le nom des classes à exécuter ?
-
-* Réponse : Fichier "JAR" (Java ARchive)
-** C'est une archive "ZIP" pour distribuer des applications en java
-** Pour exécuter le programme depuis le livrable :
-+
-[source,bash]
-----
-java -jar <fichier.jar>
-----
-
-== Exemple Java : Fabriquer un JAR (Artisanalement) 🏺 🙈
-
-* Il faut un fichier descripteur appelé `MANIFEST.mf` :
-+
-[source]
-----
-Main-Class: HelloWorld
-Class-Path: target/
-----
-
-* Il faut utiliser la commande `jar` pour fabriquer l'archive puis la tester :
-+
-[source,bash]
-----
-jar --create --manifest=MANIFEST.mf --file helloworld.jar target/HelloWorld.class
-java -jar helloworld.jar
-----
-
-== Exemple réel "à la main" Java -> JAR 🤲🏽
-
-* Essayons avec un vrai projet : reprenons le projet `menu-server`
-
-* Essayez de le compiler avec `javac` :
-+
-[source,bash]
-----
-javac ./src/main/java/com/cicdlectures/menuserver/MenuServerApplication.java -d ./target
-----
-+
-[source]
-----
-# ...
-./src/main/java/com/cicdlectures/menuserver/MenuServerApplication.java:3: error: package org.springframework.boot does not exist
-import org.springframework.boot.SpringApplication;
-                               ^
-# ...
-----
-
-🙈 Problème de dépendances
-
-== Checkpoint 🎯
-
-=> C'est vite **complexe** alors qu'on veut fabriquer / tester un programme
-
-== Do It Ourselves or Reinvent the Wheel ?
-
-*Problème* :
-[.small]
-Doit-on redéfinir tout ce processus à chaque fois ?
-
-image::reinvent-wheel.jpg[width=20%]
-
-*Réponse(s)* :
-
-[.small]
-* Dans la vraie vie, ça dépends donc ce sera à vous de juger et de ré-évaluer
-* Pour ce cours: 🛠 il faut un outil pour gérer le cycle de vie technique (Build -> Test -> etc.)
-
-== Introduction à Maven
+== Maven
 
 Say Hello to "Maven":
```

Comment:
Suggestion: While I agree with removing the previous part about building a jar by hand, I'm under the impression that it weakens the message about the necessity of a tool like Maven (ie: reproducible builds). How about adding one or two slides that explains how a jar is built and why we need to automate that process to make easy to reproduce?

Reply:
make sense, I'll work on thgis tomorrow then

Reply:
Changed to emphasize the need for Maven. WDYT?